app/services/bulk_affiliate_deactivate_uploader.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+class BulkAffiliateDeactivateUploader < BulkUploaderBase
+  def initialize(filename, filepath)
+    super(filename, filepath, nil)
+  end
+
+  private
+
+  def process_row(row)
+    affiliate_id_text = extract_identifier_from_row(row)
+
+    if affiliate_id_text.present?
+      unless affiliate_id_text.match?(/^\d+$/)
+        @results.add_failure(affiliate_id_text, 'Invalid format: Affiliate ID must be numeric.')
+        logger.warn "Skipping row with non-numeric Affiliate ID '#{affiliate_id_text}' in #{self.class.name} for file: #{@file_name}"
+        return
```

Comment:
Using return to stop execution is a bad practice. Please use `if` `else` statements instead. Perhaps have a function that validates the id and continue with the deactivation or else process errors.